### ISSUE TITLE
Visualisation convenience functions

### ISSUE THREAD

**Issue by jgroehl** [IMSY-DKFZ/simpa#4]
- [x] Data axis selection should not update any of the 3D visualization plots
~~- [ ] Contour 3D seems to fail for many volumes, and it is slow computations, it could be replaced by  dash_slicer https://dash.plotly.com/slicer~~
- [x] Volume slicer should not modify any of the 3D visualizations
- [x] Changing color scale should not update 3D plots
- [x] Right side slider bottom controller does not seem to update plot, should be fixed
- [x] Color scale should update colors in contour 2D
- [x] Spectral Visualizations should autoscale after each new point is added
- [x] Dimensions of volume should rotate along with volume when axis changes in dropdown menu
- [x] Reset points button should reset the layout of the spectral visualization scatter plot
- [x] Remove duplicated callbacks that update plots `plot_11` and `plot_12`
- [ ] Add `reconstruction` parameters to the dropdown menus so that they can also be visualized
- [x] Visualization tabs should only include plots and not data selection columns
- [x] Tabs should be mutually exclusive. When one is activated, the other is deactivated
- [x] Replace data selection axis from dropdown to radio buttons
- [x] Volume slicer should update after selecting a different axis 
- [x] Add interactions to select regions on plot_11 and show histogram on Spectral visualization
- [x] Use segmentation keywords to create color scale with names
- [x] Fix error `SVGGeometryElement.getPointAtLength:No path available for measuring`
- [x] Show annotations when clicking on data `plot_11` and `plot_12`
- [x] Sync `param1` with `param3`
- [x] Add number of bins controller for histogram plot
- [x] Replace annotation button with switch
- [x] Deactivate n_bins controller when annotations switch is off
- [ ] Add documentation to each callback
- [ ] Write unit tests for main callbacks
- [ ] Data fields should be read from SIMPA output by looking at data fields that start with `data_field`
- [ ] App should support reading SIMPA outputs that contain both 3D and 2D volumes as well as time series data
~~- [ ] Disable volume axis radio buttons if the volume is only 2D~~

A feature is needed that adds convenience functions for data visualisation.
Currently, a script is present in the simpa_examples package that does this and there is some work that is being done on creating a dash visualisation script.

I propose to add a **visualise** package in SIMPA that contains such matplotlib or dash/plotly visualisations.

**Comment by jgroehl** [comment]
added visualise module and moved the matplotlib visualisation script here.

**Comment by jgroehl** [comment]
@leoyala I have added a visualisation module into the current develop where you can integrate the dash app :)

**Comment by jgroehl** [comment]
As soon as #57 is merged into develop and master, it would be a good time to reintegrate the dash app.

**Comment by jgroehl** [comment]
I still believe it is worth including the dash app in our 1.0.0 release if it is usable to some extent.
What do you think @kdreher, @leoyala ?

**Comment by leoyala** [comment]
It is definitely in a state that is usable. I just need to fix something related to the reading acoustic properties from the HDF5 files. Showing optical properties works 😄

**Comment by leoyala** [comment]
The following should be fixed before release:

- [ ] Data axis selection should not update any of the 3D visualization plots
- [ ] Contour 3D seems to fail for many volumes, and it is slow computations, it could be replaced by  dash_slicer https://dash.plotly.com/slicer
- [ ] Volume slicer should not modify any of the 3D visualizations
- [ ] Changing color scale should not update 3D plots
- [ ] Right side slider bottom controller does not seem to update plot, should be fixed
- [ ] Color scale should update colors in contour 2D
- [ ] Spectral Visualizations should autoscale after each new point is added

**Comment by leoyala** [comment]
The following could be added to include reconstructions in the app
```data_fields['reconstructed_data'] = dict()
    for wavelength in data.wavelengths:
        data_fields['reconstructed_data'][wavelength] = get_data_field_from_simpa_output(data.simpa_output, 'reconstructed_data', wavelength)```

**Comment by leoyala** [comment]
#103 needs to be finished before we can release the Dash app

**Comment by leoyala** [comment]
After updating how data fields are extracted from `simpa_output` I get the following error. This is due to the fact that some fields are 2D and some are 3D, this needs to be fixed. 

```
Traceback (most recent call last):
  File "/home/leo/DKFZ/simpa/simpa/visualisation/dash_simulations.py", line 969, in plot_spectrum
    spectral_values.append(array[y, x])
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

**Comment by leoyala** [comment]
One possible new feature could be to include `dash_vtk` for volume rendering. 
[CommunityPost dash_vtk](https://community.plotly.com/t/announcing-dash-vtk-for-3d-simulation-graphics/49866)
